lib/native/auth.ts

```diff
@@ -80,6 +80,47 @@ export interface CognitoCredentialsProviderConfig {
     httpProxyOptions?: HttpProxyOptions;
 }
 
+/**
+ * Definition for the configuration needed to create a X509-based Credentials Provider
+ *
+ * @category Auth
+ */
+export interface X509CredentialsConfig {
+
+    /**
+     * X509 service regional endpoint to source credentials from.
+     * This is a per-account value that can be determined via the CLI:
+     * `aws iot describe-endpoint --endpoint-type iot:CredentialProvider`
+     */
+    endpoint: string;
+
+    /**
+     * The name of the IoT thing to use to fetch credentials.
+     */
+    thing_name: string;
+
+    /**
+     * The name of the role alias to fetch credentials through.
+     */
+    role_alias: string;
+
+    /**
+     * TLS context for secure socket connections.
+     * If undefined, then a default tls context will be created and used.
+     */
+    tlsContext?: ClientTlsContext;
+
+    /**
+     * Client bootstrap to use.  In almost all cases, this can be left undefined.
+     */
+    bootstrap?: ClientBootstrap;
```

Comment:
I wouldn't even bother with this and just use the default inside the native code.